### PR TITLE
Default case for switch.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: src/encoder.rs [hunk 1]
```diff
@@ -204,17 +204,37 @@ impl<InstructionsType: Instructions> Encoder<InstructionsType> {
...
 
             self.encode_block(body);
 
-            let skip_condition = SMTExpression::Not(Box::new(SMTExpression::Eq(
-                Box::new(discriminator[0].name.clone()),
-                self.encode_literal_value(literal.as_ref().unwrap()),
-            )));
+            let skip_condition = if is_default {
+                format!(
+                    "(or {})",
```
Comment (leonardoalt) [inline src/encoder.rs:215]: why or?

Comment (chriseth) [reply src/encoder.rs:215]: It's the "skip condition". For every single case it's `(not (eq <discriminator> <case label>))`. So the default is skipped whenever one of the cases is taken. This means `(or (eq <discriminator> <case label>)...)`

Comment (chriseth) [reply src/encoder.rs:215]: Ah, should test with a single default case...

**Comment by chriseth** [comment]
Yeah, isn't switch with only a default case even illegal yul? Let's do without the test...

**Comment by leonardoalt** [comment]
is it? ok

**Pull request merged**